### PR TITLE
Ensure YouTube video embeds don't set tracking cookies

### LINKED ISSUE #260: Create a no-cookie embed option for YouTube embeds
Acceptance criteria:
- [x] When a user pastes in a URL for a YouTube link, the URL we actually either store or at least render in the page should be on the youtube-nocookie.com domain
- [x] This behaviour must be available as a new/distinct [embed finder](https://docs.wagtail.org/en/stable/advanced_topics/embeds.html#configuring-embed-finders)

### PR DESCRIPTION
**Pull request by stevejalim**
## Description

We currently have configured Birdbox to use Oembed to embed YouTube and Vimeo videos into a page, via a custom `VideoBlock`. Vimeo does not set tracking cookies, but the default YouTube embed does. 

This changeset updates our YouTube embedding to use a special embed 'finder' that rewrites the oembed markup:

* instead of youtube.com we use youtube-nocookie.com
* we use trimmed back markup that we control for the iframe, so that only essential attributes are included

- [X] I have manually tested this.
- [X] I have recorded this change in `CHANGELOG.md`.

### Issue

Resolves #260

### Testing and points to consider

* Check out this branch
* Edit a page that allows adding a VideoBlock (eg a `Blog Post` or a `General Purpose Page`) 
* Add a link to a YT video you have not embedded before (because Wagtail caches old embed data in the database)
* Publish the page and view it, checking the video works and can be played in fullscreen
* In devtools: 
  * confirm that the video is loaded from the youtube-nocookie.com domain
  * confirm no cookie (or a pending consent cookie) is all that is set from that domain
* Do you agree that no <p> tag is needed inside the `iframe` (which is what we do on mozorg - see https://github.com/mozilla/bedrock/issues/14042)? If not, can you explain why it's there as it triggers HTML validation failure.

### CODE REVIEW

**Comment by stevejalim** [comment]
Can I give this one a bit of a bump my dudes?

**Review by robhudson — APPROVED** [review]
Nice clear code. r+

File: birdbox/common/embed.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    def _get_cookieless_embed_html(self, embed_html, title, original_url):
+        """Replace the oembed-generated iframe with one we control more,
+        and ensure it's using the youtube-nocookie.com domain"""
+
+        # NB: no try/except here. This is called during the page building/editing
+        # step, so we want this to fail hard, to stop bad embeds being used/published
+
+        # We could work out the embed URL from the original, but it's in the iframe,
+        # so we might as well get it and reuse it in the replacement iframe contnent
```
Comment (stevejalim) [inline birdbox/common/embed.py:42]: Typo fix
```suggestion
        # so we might as well get it and reuse it in the replacement iframe content
```

**Review by stevejalim — COMMENTED** [review]
Self r+

**Pull request merged**